plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/ClickHouseClient.java

```diff
@@ -271,6 +273,43 @@ public boolean supportsAggregationPushdown(ConnectorSession session, JdbcTableHa
         return preventTextualTypeAggregationPushdown(groupingSets);
     }
 
+    @Override
+    public boolean supportsTopN(ConnectorSession session, JdbcTableHandle handle, List<JdbcSortItem> sortOrder)
+    {
+        for (JdbcSortItem sortItem : sortOrder) {
+            Type sortItemType = sortItem.column().getColumnType();
+            if (sortItemType instanceof VarcharType) {
+                return false;
+            }
+
+            if (sortItemType instanceof CharType) {
+                throw new TrinoException(GENERIC_INTERNAL_ERROR, "Unexpected column type CharType");
```

Comment:
We use `TrinoException` when the exception is supposed to be reached to users. We can use `checkArgument` or `checkState` instead:
```java
            Type sortItemType = sortItem.column().getColumnType();
            checkArgument(!(sortItemType instanceof CharType), "Unexpected char type: %s", sortItem.column().getColumnName());
            if (sortItemType instanceof VarcharType) {
                return false;
            }
```

Reply:
Fixed, thanks.